rex/joint_pd/joint_pd.py

```diff
@@ -0,0 +1,413 @@
+# -*- coding: utf-8 -*-
+"""
+General Joint Probabilty Distribution calculator
+"""
+from concurrent.futures import as_completed
+import gc
+import logging
+import h5py
+import numpy as np
+import os
+import pandas as pd
+from warnings import warn
+
+from rex.version import __version__
+from rex.resource import Resource
+from rex.utilities.execution import SpawnProcessPool
+from rex.utilities.loggers import log_mem, log_versions
+from rex.utilities.utilities import slice_sites, to_records_array
+
+logger = logging.getLogger(__name__)
+
+
+class JointPD:
```

Comment:
For joint dists i find a plotting utility to be SUPER helpful.

Reply:
What sort of plot do you like?

Reply:
https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.hist2d.html